### ISSUE TITLE
MemoryFileSystem::newOutputFileStream does not check the path it opens for write permissions

### ISSUE THREAD

**Issue by stwZiemer** [marschall/memoryfilesystem#138]
I think that `MemoryFileSystem::newOutputFileStream` does not check the path it opens for write permissions.
I can reproduce the issue in the test class below. Is there something I'm missing?

I used the most current version (2.4.0)

```java
package com.github.marschall.memoryfilesystem;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.AccessDeniedException;
import java.nio.file.FileSystem;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.attribute.PosixFileAttributeView;
import java.util.Collections;

class WritePermissionTest {
    @Test
    void outputStreamChecksPermissions() throws IOException {
        try (FileSystem fileSystem = MemoryFileSystemBuilder.newLinux().build()) {
            final Path path = fileSystem.getPath("/a.csv");
            Files.createFile(path);

            PosixFileAttributeView fileAttributeView = fileSystem.provider().getFileAttributeView(path, PosixFileAttributeView.class);
            fileAttributeView.setPermissions(Collections.emptySet());

            // double check that writing not allowed
            // check passed
            Assertions.assertFalse(Files.isWritable(path));

            // should throw in my opinion but doesn't
            Assertions.assertThrows(IOException.class, () -> Files.newOutputStream(path));
        }
    }
}

```

**Comment by marschall** [comment]
I just released 2.5.0 which contains a fix.

**Comment by stwZiemer** [comment]
Thank you very much! We’ve found many bugs thanks to your memory file system.

Von: Philippe Marschall ***@***.***>
Gesendet: Dienstag, 7. Februar 2023 07:09
An: marschall/memoryfilesystem ***@***.***>
Cc: Weidmann Stefano ***@***.***>; Author ***@***.***>
Betreff: Re: [marschall/memoryfilesystem] MemoryFileSystem::newOutputFileStream does not check the path it opens for write permissions (Issue #138)


I just released 2.5.0 which contains a fix.

**Comment by marschall** [comment]
That's great to hear.
